lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -57,10 +59,10 @@ def serializable_hash_for_collection(serializer, options)
           hash
         end
 
-        def serializable_hash_for_single_resource(serializer, options)
+        def serializable_hash_for_single_resource(options)
           primary_data = primary_data_for(serializer, options)
           relationships = relationships_for(serializer)
-          included = included_for(serializer)
+          included = included_resources(@include_tree)
```

Comment:
My concern is not that much towards performance than clarity: it's easy to define a local variable with the same name as a getter/setter. Plus `@var` makes it really clear where to look for when something is wrong with `@var`. I guess it's a tradeoff.